### PR TITLE
Bug 1874584: UPSTREAM: <carry>: retry etcd errors

### PR DESCRIPTION
**Pull request by tkashem**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/etcd3retry/retry_etcdclient.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+func (c *retryClient) GuaranteedUpdate(
+	ctx context.Context, key string, ptrToType runtime.Object, ignoreNotFound bool,
+	precondtions *storage.Preconditions, tryUpdate storage.UpdateFunc, suggestion ...runtime.Object) error {
+	return onError(defaultRetry, isRetriableEtcdError, func() error {
+		return c.Interface.GuaranteedUpdate(ctx, key, ptrToType, ignoreNotFound, precondtions, tryUpdate, suggestion...)
+	})
+}
+
+func isRetriableEtcdError(err error) bool {
+	if err == nil {
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/etcd3retry/retry_etcdclient.go:153]: if the err could be read as `rpctypes.Error(err)` (https://github.com/etcd-io/etcd/blob/master/etcdserver/api/v3rpc/rpctypes/error.go#L230)  then according to https://github.com/etcd-io/etcd/blob/master/clientv3/retry.go#L53 all immutable errors that map to `codes.Unavailable` are safe to retry

**Comment by tkashem** [comment]
work on https://github.com/openshift/kubernetes/pull/322 has moved to this PR.

**Comment by deads2k** [comment]
infra

/retest

**Comment by tkashem** [comment]
Needed to ensure that there was no wait after the last retry.

/hold cancel

**Comment by p0lyn0mial** [comment]
we are going to remove this carry patch. the real issue turned out to be https://github.com/openshift/kubernetes/pull/959

**Pull request merged**